test/OpenTelemetry.Tests/Logs/LoggerProviderBuilderExtensionsTests.cs

```diff
@@ -0,0 +1,185 @@
+// <copyright file="LoggerProviderBuilderExtensionsTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Resources;
+using Xunit;
+
+namespace OpenTelemetry.Logs.Tests;
+
+public sealed class LoggerProviderBuilderExtensionsTests
+{
+    [Fact]
+    public void LoggerProviderBuilderAddInstrumentationTest()
+    {
+        List<object>? instrumentation = null;
+
+        using (var provider = Sdk.CreateLoggerProviderBuilder()
+            .AddInstrumentation<CustomInstrumentation>()
+            .AddInstrumentation((sp, provider) => new CustomInstrumentation() { Provider = provider })
+            .AddInstrumentation(new CustomInstrumentation())
+            .Build() as LoggerProviderSdk)
+        {
+            Assert.NotNull(provider);
+
+            Assert.Equal(3, provider.Instrumentations.Count);
+
+            Assert.Null(((CustomInstrumentation)provider.Instrumentations[0]).Provider);
+            Assert.False(((CustomInstrumentation)provider.Instrumentations[0]).Disposed);
+
+            Assert.NotNull(((CustomInstrumentation)provider.Instrumentations[1]).Provider);
+            Assert.False(((CustomInstrumentation)provider.Instrumentations[1]).Disposed);
+
+            Assert.Null(((CustomInstrumentation)provider.Instrumentations[2]).Provider);
+            Assert.False(((CustomInstrumentation)provider.Instrumentations[2]).Disposed);
+
+            instrumentation = new List<object>(provider.Instrumentations);
+        }
+
+        Assert.True(((CustomInstrumentation)instrumentation[0]).Disposed);
+        Assert.True(((CustomInstrumentation)instrumentation[1]).Disposed);
+        Assert.True(((CustomInstrumentation)instrumentation[2]).Disposed);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void LoggerProviderBuilderNestedResolutionUsingBuilderTest(bool callNestedConfigure)
+    {
+        bool innerTestExecuted = false;
+
+        using var provider = Sdk.CreateLoggerProviderBuilder()
+            .ConfigureServices(services =>
+            {
+                if (callNestedConfigure)
```

Comment:
Do we not want to assert something specific based on `callNestedConfigure`?

Reply:
I wrote that test so long ago I can't remember, TBH. I added an assert in the logging one and its cousins in tracing & metrics. I figured it couldn't hurt 😄